homeassistant/components/nuki/binary_sensor.py

```diff
@@ -33,13 +33,9 @@ async def async_setup_entry(
 class NukiDoorsensorEntity(NukiEntity, BinarySensorEntity):
     """Representation of a Nuki Lock Doorsensor."""
 
+    _attr_has_entity_name = True
     _attr_device_class = BinarySensorDeviceClass.DOOR
 
-    @property
-    def name(self):
-        """Return the name of the lock."""
-        return self._nuki_device.name
```

Comment:
As the doorsensor is [not the main function](https://developers.home-assistant.io/docs/core/entity/#example-of-a-switch-entity-which-is-either-not-the-main-feature-of-a-device-or-is-not-part-of-a-device) it should probably include the name property with `return "Doorsensor"`

Reply:
I tested and the code as it is already returns the name of the device, followed by "Doorsensor"

Reply:
I've just checked out your PR and got the following result: 
![image](https://user-images.githubusercontent.com/9799276/194899280-7ca83686-47db-4fcd-a366-fede2bcaadec.png)
doorsensor should start with a capital letter, as per the documentation: https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations

> Entity names should start with a capital letter, the rest of the words are lower case (unless it's a proper noun or a capitalized abbreviation of course).

Reply:
If I set a name, for example Door sensor, the entity get named "Door sensor" so naming style is wrong.
Without a defined name, I get Device name doorsensor, it should be Doorsensor you're right, I missed it.
But as I don't set the name it is something auto-generated, so should I set an entity description instead of the name?

Reply:
I think `_attr_name = "Doorsensor"` should be fine. I see, that it's only Doorsensor at the moment, that's because the nuki integration has no device registration yet. That's what my PR is for: https://github.com/home-assistant/core/pull/79806 

I've just tried your commit on my PR and it would look like this:
![image](https://user-images.githubusercontent.com/9799276/194906231-93303ab4-7f19-4ded-ab18-0dfd90ba1cbd.png)